app/pages/home-for-user/recent-projects.jsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import HomePageSection from './generic-section';
+import { Link } from 'react-router';
+import ProjectCard from '../../partials/project-card';
+import apiClient from 'panoptes-client/lib/api-client';
+
+const RecentProjectsSection = React.createClass({
+  propTypes: {
+    onClose: React.PropTypes.func,
+  },
+
+  contextTypes: {
+    user: React.PropTypes.object.isRequired,
+  },
+
+  getInitialState() {
+    return {
+      loading: false,
+      error: null,
+      projects: [],
+      avatars: {},
+    };
+  },
+
+  componentDidMount() {
+    this.fetchProjects(this.context.user);
+  },
+
+  componentWillReceiveProps(nextProps, nextContext) {
+    if (nextContext.user !== this.context.user) {
+      this.fetchProjects(nextContext.user);
+    }
+  },
+
+  fetchProjects(user) {
+    this.setState({
+      loading: true,
+      error: null,
+      avatars: {},
+    });
+
+    user.get('project_preferences', {
+      sort: '-updated_at',
+    })
+    .then((preferences) => {
+      preferences = preferences.filter((projectPreference) => {
```

Comment:
Reassign the function param seems it could cause confusion. IMO, would be better to declare a separate variable and assign the filter to that.